### PR TITLE
[k8s]: switching to Flannel from Calico.

### PR DESCRIPTION
**Pull request by renukamanavalan**
This helps bring down the image size by around 500+MB.

**- Why I did it**

**- How I did it**
Privately building with INSTALL_KUBERNETES=y to test

**- How to verify it**
Build it with INSTALL_KUBERNETES=y

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by lguohan** [comment]
retest vsimage please

File: files/build_templates/sonic_debian_extension.j2 [hunk 1]
```diff
@@ -462,11 +462,10 @@ sudo LANG=C chroot $FILESYSTEM_ROOT docker $SONIC_NATIVE_DOCKERD_FOR_DOCKERFS ta
 {% if install_kubernetes == "y" %}
 ## Pull in kubernetes docker images
 echo "pulling universal k8s images ..."
+K8S_ARCH=$([ "${CONFIGURED_ARCH}" == "armhf" ] && echo "arm64" || echo "${CONFIGURED_ARCH}")
```
Comment (qiluo-msft) [inline files/build_templates/sonic_debian_extension.j2:465]: >K8S_ARCH [](start = 0, length = 8)

Is it FLANNEL_ARCH ?

**Comment by qiluo-msft** [comment]
> This has a big gain on image size.

Is it a big gain or big decreasing?

File: rules/config [hunk 2]
```diff
@@ -137,13 +137,13 @@ INSTALL_KUBERNETES = n
...
 # KUBERNETES_VERSION - Set to the required version.
 # K8s_GCR_IO_PAUSE_VERSION - Version of k8s universal pause container image
-# K8s_CNI_CALICO_VERSION - Calico used as CNI; Appropriate version for this Kubernetes version
+# K8s_CNI_FLANNEL_VERSION - Flannel used as CNI; Appropriate version for this Kubernetes version
 # These are Used *only* when INSTALL_KUBERNETES=y
 # NOTE: As a worker node it has to run version compatible to kubernetes master.
 #
 KUBERNETES_VERSION = 1.18.0
 K8s_GCR_IO_PAUSE_VERSION = 3.2
-K8s_CNI_CALICO_VERSION = 3.12.1
```
Comment (qiluo-msft) [inline rules/config:146]: >K8s_CNI [](start = 0, length = 7)

Do we need to make K8s_CNI configurable between calico/flannel, or others?

Comment (renukamanavalan) [reply rules/config:146]: May be. But we will wait before we do that. We need to collect more data/requirements, before we make this call.

File: files/build_templates/sonic_debian_extension.j2 (hunk 1 above)
Comment (renukamanavalan) [reply files/build_templates/sonic_debian_extension.j2:465]: Guess, yes for now.  Do you want this changed ? I could.

Comment (qiluo-msft) [reply files/build_templates/sonic_debian_extension.j2:465]: Yes, please fix it.

---
In reply to: [439624555](https://github.com/Azure/sonic-buildimage/pull/4768#discussion_r439624555) [](ancestors = 439624555)

Comment (renukamanavalan) [reply files/build_templates/sonic_debian_extension.j2]: done

**Pull request merged**